lib/config/configuration.go

```diff
@@ -81,6 +81,9 @@ type CommandLineFlags struct {
 	GopsAddr string
 	// DiagnosticAddr is listen address for diagnostic endpoint
 	DiagnosticAddr string
+	// PermitUserEnvironment enables reading of ~/.tsh/environment
+	// when creating a new session.
+	PermitUserEnvironment bool
```

Comment:
I would not make this a boolean flag, rather than a list with files to read with default to `.tsh/environment` so folks can actually set to something else if they want, e.g. will close the issue #1011